public/app/features/datasource-drawer/DataSourcePickerWithHistory.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import { dateTime } from '@grafana/data';
+import { LocalStorageValueProvider } from 'app/core/components/LocalStorageValueProvider';
+
+import { DataSourcePicker } from './DataSourcePicker';
+import { DataSourcePickerHistoryItem, DataSourcePickerWithHistoryProps } from './types';
+
+const DS_PICKER_STORAGE_KEY = 'DATASOURCE_PICKER';
+
+export const DataSourcePickerWithHistory = (props: DataSourcePickerWithHistoryProps) => {
+  return (
+    <LocalStorageValueProvider<DataSourcePickerHistoryItem[]>
+      defaultValue={[]}
+      storageKey={props.key ?? DS_PICKER_STORAGE_KEY}
```

Comment:
`key` is misleading for React developers we shouldn't use it. could we rename it to `localStorageKey` instead?

Reply:
Good point